src/shogun/lib/parameter_observers/ParameterObserver.h

```diff
@@ -87,12 +88,42 @@ namespace shogun
 		 */
 		virtual bool filter(const std::string& param);
 
+		/**
+		 * Get the i-th observation recorded. It will be presented
+		 * to the user as an Any value which need to be casted.
+		 * @param i id of the observation
+		 * @return Any value with the observation
+		 */
+		virtual Any get_observation(int i);
```

Comment:
Users should never receive an any, it is internal. Rather, they can access parameters with get

Reply:
See above discussion.